config/sql/config.go

```diff
@@ -34,6 +66,16 @@ func Configure(p *config.Provider) {
 				Type: rconfig.APISPackagePath + "/azure/v1alpha1.ResourceGroup",
 			},
 		}
+		r.Sensitive.AdditionalConnectionDetailsFn = msSQLConnectionDetails
+		r.UseAsync = true
+	})
+	p.AddResourceConfigurator("azurerm_mssql_server", func(r *config.Resource) {
+		r.References = config.References{
```

Comment:
```suggestion
		r.Group = "sql"
		r.References = config.References{
```

Should this be in sql group since the service name is `Microsoft.Sql`?

Reply:
Because we are generating both sql and mssql, we can keep their groups separated. Or we can have sql.MicrosoftSQL and sql.SQL. But I prefer to stay close to Terraform naming.

Reply:
My preference is usually stick with the cloud provider itself but both woudl work as long as it's consistent across the repo.